listenbrainz/db/model/review.py

```diff
@@ -0,0 +1,46 @@
+import sqlalchemy
+from pydantic import BaseModel
+from typing import List
+
+from listenbrainz.db import timescale
+
+
+class CBReviewMetadata(BaseModel):
+    """ Model to represent the review payload sent to the CB api
+    Some fields is_draft and license_choice are added by the backend
+    and always the same. Hence, omitted from this model.
+    """
+    name: str
+    entity_type: str
+    rating: int
+    text: str
+    entity_id: str
+    recording_id: str
+    language: str
+
+
+class CBReviewTimelineMetadata(BaseModel):
+    """Model to represent the data stored in user timeline event table
+    for a CB review. We only store review uuid, the entity's mbid and
+    the recording mbid (of the listen using which the review was written).
+    For the review of a recording, the entity_id and the recording_id is
+    the same. Other data is retrieved from the CB api as needed.
+    """
+    entity_name: str
+    review_id: str
+    entity_id: str
```

Comment:
It is possible to get away with a subset of these fields here but I decided to keep all for simpler implementation. If we needed to support recording reviews only, we only need to store the recording mbid  and review id  here.  The entity name is obtained from the mapping tables when needed. But because we support artist and release group reviews, we need to store the name field because we don't have those in the mapping tables. 

Now for making these artists and release group reviews playable we could just search for the names. Alternative is we attach a recording mbid with each review and then retrieve the recording data from mapped tables and play that listen for the review. Regardless of the approach we go with I think its a good idea to store a recording mbid with each review. The reason is eveyrthing else in LB has a msid /mbid attached to it. Therefore, IMO It makes sense for reviews to be that way as well.

Now for recording reviews, the entity_id and entity_name become redundant because the entity_id is same as recording_id and entity_name can be retrieved from mapping tables. So we can set these fields to null if we want for recording reviews. I decided in favor of filling the redundant fields anyways because feels simpler to me.

Reply:
I removed recording_id after discussion on IRC. We don't have a use case for this and will need to add more stuff to handle it.